compiler/qsc_hir/src/global.rs

```diff
@@ -157,6 +157,17 @@ impl PackageIter<'_> {
                     intrinsic: decl.body.body == SpecBody::Gen(SpecGen::Intrinsic),
                 }),
             }),
+            (ItemKind::Callable(decl), None) => Some(Global {
```

Comment:
If I understand this correctly, these include items declared in local scopes. These aren't really "globals" though, are they? Bit concerned about what assumptions they're breaking elsewhere. If nothing else, I think it's misleading to call them globals.

Reply:
It actually wont. It includes callables whose direct parents are namespaces or callables that have no direct parent (defined in the global scope). Rather cleverly, our compiler generates callables lifted from within other callables (including lambdas) as having the parent be the callable item in which they were defined, so those get filtered out by this logic:
![image](https://github.com/user-attachments/assets/0066225c-d125-4b57-9e17-5bcaca9f0d33)

Reply:
> callables that have no direct parent (defined in the global scope)

I guess this is breaking my brain. Callables can't be defined in the global scope, can they?

Oh, based on the screenshot I think you're referring to the "persistent local scope"? I guess, I mean... is it okay to call them globals? I can convince myself it's harmless, but it _is_ introducing a concept we didn't use to have (global items at the top-level). I still think it's muddying the concepts a bit in the compiler internals.

Reply:
I don't quite agree... yes this is a change in that the global iterator now picks up items that are defined in the top level of a notebook cell where previously it didn't. I would say this is actually fixing a subtle bug we didn't realize we had but that never made a difference before. The global iterator would only show callables that were members of a namespace, which means when you iterated the open package for a notebook you wouldn't see any callables that were defined as top-level statements, only those defined in namespaces (which are supported in cells but discouraged because it has weird duplicate definition behaviors). The reason this didn't end up mattering is that we only used `global::iter_package` in a few places that occur before notebook compilation comes into the picture:
 - When constructing a new package store from an existing compile unit (compiler/qsc_frontend/src/compile.rs line 227)
 - When adding an external package dependency to a compilation (compiler/qsc_frontend/src/resolve.rs line 1621)
 - When running passes on the core library (compiler/qsc_passes/src/lib.rs line 159)

It just so happens that all these never need to run on items defined in a notebook cell without a namespace, so we never noticed they were missing from the iteration. Adding the new case to the iteration doesn't affect any of these locations because none of those run on packages that items defined outside of a namespace.

Reply:
OK... I've actually changed my mind, you're right. Your example from the side chat is a pretty convincing example demonstrating how these are actual globals:

![image](https://github.com/user-attachments/assets/a90193e3-83e9-4710-a17a-faf88ca66238)